kayenta-core/src/main/java/com/netflix/kayenta/canary/providers/InfluxdbCanaryMetricSetQueryConfig.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Joseph Motha
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.kayenta.canary.providers;
+
+import java.util.List;
+
+import javax.validation.constraints.NotNull;
+
+import com.netflix.kayenta.canary.CanaryMetricSetQueryConfig;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Builder
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class InfluxdbCanaryMetricSetQueryConfig implements CanaryMetricSetQueryConfig {
+  
+  @NotNull
+  @Getter
+  private String metricName;
+  
+  @Getter
+  private List<String> fields;
+  
+  //@Getter
+  //private List<String> groupByFields;
```

Comment:
You already have the other `TODO` referencing `groupByFields`. I'd drop the commented-out code.